### ISSUE TITLE
Error: ENOENT: no such file or directory, open 'C:\Users\kudo\01_Work\Repos\expo\expo\android\versioned-react-native

### ISSUE THREAD

**Issue by shop-fluencer** [expo/expo#20305]
### Summary

I have upgraded expo from 46 to 47 and added hermes on app.json

now I get this error message

```
Error: ENOENT: no such file or directory, open 'C:\Users\kudo\01_Work\Repos\expo\expo\android\versioned-react-native\ReactAndroid\hermes-engine\.cxx\MinSizeRel\2q85j1u4\arm64-v8a\lib\InternalBytecode\InternalBytecode.js'
    at Object.openSync (node:fs:585:3)
    at Object.readFileSync (node:fs:453:35)
    at getCodeFrame (C:\xxx\node_modules\metro\src\Server.js:1028:18)
    at Server._symbolicate (C:\xxx\node_modules\metro\src\Server.js:1101:22)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Server._processRequest (C:\xxx\node_modules\metro\src\Server.js:458:7) {
  errno: -4058,
  code: 'ENOENT',
  path: 'C:\\Users\\kudo\\01_Work\\Repos\\expo\\expo\\android\\versioned-react-native\\ReactAndroid\\hermes-engine\\.cxx\\MinSizeRel\\2q85j1u4\\arm64-v8a\\lib\\InternalBytecode\\InternalBytecode.js'
}
```

This happend when I dispatch anything with RTK Query

Only happens on Android.

### What platform(s) does this occur on?

Android

### Environment

expo-env-info 1.0.5 environment info:
System:
OS: Windows 10 10.0.19044
Binaries:
Node: 16.13.2 - C:\Program Files\nodejs\node.EXE
Yarn: 1.22.11 - ~\AppData\Roaming\npm\yarn.CMD
npm: 8.1.2 - C:\Program Files\nodejs\npm.CMD
IDEs:
Android Studio: Version 2020.3.0.0 AI-203.7717.56.2031.7583922
npmPackages:
expo: ^47.0.0 => 47.0.8
react: 18.1.0 => 18.1.0
react-dom: 18.1.0 => 18.1.0
react-native: 0.70.5 => 0.70.5
react-native-web: ~0.18.7 => 0.18.8
Expo Workflow: managed

### Minimal reproducible example

```
import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/dist/query/react";
// @ts-ignore
import { API_ENDPOINT } from '@env';
import { IProduct } from "../../../components/Product";

export const WishlistApi = createApi({
  reducerPath: 'WishlistApi',
  baseQuery: fetchBaseQuery({baseUrl: `${API_ENDPOINT}/`}),
  tagTypes: ['WISHLIST', 'HOME'],
  endpoints: (build) => ({
    // Wishlist
    wishlist: build.query<(IProduct & { is_wishlist: boolean; })[], { user_id: number; }>({
      query: (data) => ({
        url: '/wishlist',
        method: 'POST',
        body: data
      }),
      providesTags: ['WISHLIST']
    }),
    // Main Products
    homeList: build.query<(IProduct & { is_wishlist: boolean; })[], void>({
      query: (data) => ({
        url: '/home',
        method: 'POST',
        body: data
      }),
//      providesTags: ['HOME']
    }),
    testList: build.query<(IProduct & { is_wishlist: boolean; })[], void>({
      query: (data) => ({
        url: '/test',
        method: 'POST',
        body: data
      }),
//      providesTags: ['HOME']
    }),
    getPost: build.query<IProduct, number>({
      query: (id) => `get_product/${id}`,
      providesTags: ['HOME'],
    }),
//      providesTags: ['HOME']
    updateHomelist: build.mutation<void, { product_id: number; product: IProduct; user_id?: number; }>({
      query: (data) => ({
        url: '/add_to_wishlist',
        method: 'POST',
        body: data
      }),
      async onQueryStarted({ product_id, product, user_id }, { dispatch, queryFulfilled }) {

        const patchResult = dispatch(
          WishlistApi.util.updateQueryData('homeList', undefined, (draft) => {
            const pf = draft?.find((el => el.id === product_id));
            if(pf) {
              pf.is_wishlist = !pf.is_wishlist;
            }
           // Object.assign(draft, product)
          })
        )
   
        const patchResult2 = dispatch(
          WishlistApi.util.updateQueryData('testList', undefined, (draft) => {
            const pf = draft?.find((el => el.id === product_id));
            if(pf) {
              pf.is_wishlist = !pf.is_wishlist;
            }
           // Object.assign(draft, product)
          })
        )

        const patchResult3 = dispatch(
          WishlistApi.util.updateQueryData('wishlist', { user_id }, (draft) => {
            console.log('HI');
            const dd = draft?.find((el => el.id === product_id));
            console.log(dd);
            if(dd) {
              return draft.filter((el => el.id !== product_id));
            } else {
              draft.push({...product, is_wishlist: true});
            }
           
           // Object.assign(draft, product)
          })
        )
        try {
          await queryFulfilled
        } catch {
          patchResult.undo();
          patchResult2.undo();
          patchResult3.undo();

          /**
           * Alternatively, on failure you can invalidate the corresponding cache tags
           * to trigger a re-fetch:
           * dispatch(api.util.invalidateTags(['Post']))
           */
        }
      },
    //  invalidatesTags: ['WISHLIST']
    }),
...
```

**Comment by Thodor12** [comment]
I am constantly getting the exact same error in SDK 46, it looks like someone committed an absolute path to load in a library (maybe metro, not quite sure) because of the hardcoded path `D:\Users\kudo\01_Work`

**Comment by Kudo** [comment]
hi there! does the issue happen on expo go + the minimal reproducible example code + running `npx expo start` on windows host?

**Comment by Thodor12** [comment]
Hey @Kudo it doesn't appear on a minimal repro because I think it has something to do with the usage of redux.
OP stated he's using RTK Query, I'm also using Redux + Redux persist in my app, in my case it's throwing the error on startup (when redux-persist wants to hydrate the state), but it does not error consistently each time.

Considering the path contains your name, have you by any chance contributed to any of redux it's libraries and may have forgotten an absolute path?

**Comment by Thodor12** [comment]
Retract that previous statement, I've done some testing and checking in that stacktrace, it does come from metro, but it appears whenever metro wants to display a warning message on the screen of the device.

That's what that _symbolize call does. If I open my project and purposefully make a compilation error, that's when the error appears.
So I put a logger on that statement like so
```js
  async _symbolicate(req, res) {
    const getCodeFrame = (urls, symbolicatedStack) => {
      for (let i = 0; i < symbolicatedStack.length; i++) {
        const { collapse, column, file, lineNumber } = symbolicatedStack[i];
        const fileAbsolute = path.resolve(
          this._config.projectRoot,
          file !== null && file !== void 0 ? file : ""
        );
        console.log(fileAbsolute, this._config.projectRoot)
```

For most files this prints correctly like this:
```
D:\Projects\IPM\App\node_modules\react-devtools-core\dist\backend.js D:\Projects\IPM\App
D:\Projects\IPM\App\node_modules\expo\build\logs\RemoteConsole.js D:\Projects\IPM\App
D:\Projects\IPM\App\node_modules\expo\build\environment\react-native-logs.fx.js D:\Projects\IPM\App
```
But there's one entry which shows this:
```
D:\Users\kudo\01_Work\Repos\expo\expo\android\versioned-react-native\ReactAndroid\hermes-engine\.cxx\MinSizeRel\6d3b5k69\arm64-v8a\lib\InternalBytecode\InternalBytecode.js D:\Projects\IPM\App
```

I have no idea from where this symbolization is being initiated, but it does look like there's some faulty path set somewhere.

**Comment by Kudo** [comment]
i still cannot repro the problem. could you create a minimal reproducible example and upload to some git repository? that would help us to further investigate the issue.

**Comment by tschai-yim** [comment]
@Kudo Have you tested it on a different machine? Because it is your absolute path it would obviously work on your normal setup.

Can you find anything by just searching for the path in all codebases? I can't imagine the path is stored in any other way.

**Comment by Kudo** [comment]
yes i did try on windows mahcine which doesn't have the same path.

well, i finally get the repro today. it's from unhandled promise rejection. the error only show in cli terminal and not impact to the app. i'll try to figure out what happen to hermes internally.

```tsx
export default function App() {
  async function foo() {
    throw new Error("unhandled promise rejection");
  }

  useEffect(() => {
    foo();
  }, []);
 // ...
}
```

**Comment by Kudo** [comment]
hi there! we've published `@expo/metro-config@0.5.2` with a fix. from a new init project, the problem should be fixed. from an existing sdk 47 project, please bump the `@expo/metro-config` inside lock file.

e.g. in `yarn.lock`, remove this block.
```
"@expo/metro-config@~0.5.0", "@expo/metro-config@~0.5.1":
  version "0.5.1"
  resolved "https://registry.yarnpkg.com/@expo/metro-config/-/metro-config-0.5.1.tgz#58c715041e1773ad653519535c017138bfc766de"
  integrity sha512-Rvy4ZFgKNDfXO401z2OQF8fWbPj1lLVDL4GF1aqCIhCDHCKrezbwB0xejpcUyndJRCxBL2BMAM+P24t6cKv9Fw==
  dependencies:
    "@expo/config" "~7.0.2"
    "@expo/json-file" "8.2.36"
    chalk "^4.1.0"
    debug "^4.3.2"
    find-yarn-workspace-root "~2.0.0"
    getenv "^1.0.0"
    resolve-from "^5.0.0"
    sucrase "^3.20.0"
```
save the file and do `yarn` installation again.

**Comment by bartvanandel** [comment]
For those who are using Expo SDK 47 and seeing this issue, manually editing package-lock.json is no longer necessary. Expo can do it for you by running `expo upgrade 47` (yes, same version) in a terminal (provided you have `expi-cli` installed).
